src/ports/realm.ts

```diff
@@ -0,0 +1,145 @@
+import { IBaseComponent } from '@well-known-components/interfaces'
+import { BaseComponents } from '../types'
+import * as fs from 'fs/promises'
+
+export const defaultNames = [
+  'zeus',
+  'poseidon',
+  'athena',
+  'hera',
+  'hephaestus',
+  'aphrodite',
+  'hades',
+  'hermes',
+  'artemis',
+  'thor',
+  'loki',
+  'odin',
+  'freyja',
+  'fenrir',
+  'heimdallr',
+  'baldr'
+]
+
+export type IRealmComponent = IBaseComponent & {
+  getName(lighthouseName?: string): Promise<string | undefined>
+}
+
+export const CATALYST_NAME_CONFIG_FILE = '.catalyst-name'
```

Comment:
What if two catalyst owners write down the same name in `.catalyst-name` file? Will it work? Should that be mitigated?

Reply:
if there is race condition you mean? yes, this is not handled, but there is no way to be 100% accurate, I think we can fix it manually if it happens, until we have a more robust mechanism

Reply:
Okay, we can live with that. Can we remove from the `defaultNames` the ones that are already taken by the DAO catalysts?

Reply:
done!